test.py

```diff
@@ -37,7 +37,7 @@
     MODES = [args.mode] if args.mode else seastar_cmake.SUPPORTED_MODES
 
     def run_tests(mode):
-        BUILD_PATH = seastar_cmake.BUILD_PATHS[mode]
+        BUILD_PATH = seastar_cmake.build_path(mode)
```

Comment:
I believe this place needs more care. If seastar is built with, say, `--build-root=build-x` then there's no chance to run tests on it as `test.py` will try to find test binaries in `build/${mode}` and will apparently fail

Reply:
You are right, I wasn't using test.py so I did not add support for non-default roots here, but I will now.

Reply:
@xemul when I say "now" I mean it loosely, say "within the next 12 months" :).

So here I am within that period finally with the change to `test.py` as well.

I've tested this locally with something like:

```
for v in 10 11 12; do ./configure.py --build-root "build-gcc-$v" --mode dev --compiler "g++-$v" --c-compiler  ~/dev/redpanda/vbuild/llvm/install/bin/clang --compile-commands-json; done
for v in 10 11 12; do ./test.py --build-root build-gcc-$v --mode dev; done
```

... and it's working fine, running the tests in each build-root as expected.